core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -363,7 +480,12 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
           // Create a new prewarm container
           // NOTE: prewarming ignores the action code in exec, but this is dangerous as the field is accessible to the
           // factory
-          prewarmContainer(action.exec, memory)
+          val ttl = data.expires match {
+            case Some(deadline) =>
+              Some(deadline.time)
+            case None => None
+          }
+          prewarmContainer(action.exec, memory, ttl)
```

Comment:
I don't think the deadline should propagate from the previous prewarm - it should be set new based on `ReactivePrewarmingConfig.ttl from now`

Reply:
I see never mind, I guess `deadline.time` is same as configured ttl?

Reply:
In that case, remove the `val ttl = data.expires match`, and just call `prewarmContainer(action.exec, memory, data.expires)`
I think you'll need to change ttl type from `Duration` to `FiniteDuration`

Reply:
I changed from runtime.json's reactive config's ttl is `FiniteDuration`

Regarding `I guess deadline.time is same as configured ttl?`
Yes

Regarding `remove the val ttl = data.expires match, and just call prewarmContainer(action.exec, memory, data.expires)`
hm, if that, need to change  `case class Start`'s third field to `Option[DeadLine]`, this lead to some test cases run failed and not easy to make them run successfully.

Reply:
OK, In that case, I think change everywhere fields/variables named "expires" that are not Deadline to names like "ttl" - otherwise it is confusing. "Expires" should be Deadline/instant and "ttl" should be FiniteDuration types.

Reply:
Aleady applied `"Expires" should be Deadline/instant and "ttl" should be FiniteDuration types.`